cmd/nodepool/azure/create.go

```diff
@@ -42,6 +48,12 @@ func NewCreateCommand(coreOpts *core.CreateNodePoolOptions) *cobra.Command {
 	cmd.Flags().BoolVar(&platformOpts.EnableEphemeralOSDisk, "enable-ephemeral-disk", platformOpts.EnableEphemeralOSDisk, "If enabled, the Azure VMs in the NodePool will be setup with ephemeral OS disks")
 	cmd.Flags().StringVar(&platformOpts.DiskStorageAccountType, "disk-storage-account-type", platformOpts.DiskStorageAccountType, "The disk storage account type for the OS disks for the VMs.")
 	cmd.Flags().StringVar(&platformOpts.SubnetID, "subnet-id", platformOpts.SubnetID, "The subnet id where the VMs will be placed.")
+	cmd.Flags().StringVar(&platformOpts.ImageID, "image-id", platformOpts.ImageID, "The Image ID to boot the VMs with.")
```

Comment:
Please create an `AzureMarketplaceOptions` struct, its own `BindOpts()` and use composition to embed it into the nodepool command. See how the KubeVirt node pool options are embedded into the KubeVirt cluster options for an example.

Reply:
This should be resolved now.